roles/deploy-aws-infra/tasks/main.yml

```diff
@@ -0,0 +1,343 @@
+    rules:
+      - proto: udp
+        from_port: 8053
+        to_port: 8053
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 8053
+        to_port: 8053
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 8443
+        to_port: 8443
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 443
+        to_port: 443
+        cidr_ip: 0.0.0.0/0
+      - proto: udp
+        from_port: 53
+        to_port: 53
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 53
+        to_port: 53
+        cidr_ip: 0.0.0.0/0
+      - proto: udp
+        from_port: 4789
+        to_port: 4789
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ocp_master_sg
+
+- name: Create OCP Node Security Group for the new VPC
+  ec2_group:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "labs-ocp-node"
+    description: "Labs OCP Node"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+    rules:
+      - proto: udp
+        from_port: 4789
+        to_port: 4789
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 10250
+        to_port: 10250
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ocp_node_sg
+
+- name: Create OCP Infra Security Group for the new VPC
+  ec2_group:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "labs-ocp-infra"
+    description: "Labs OCP Infra"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+    rules:
+      - proto: tcp
+        from_port: 80
+        to_port: 80
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 443
+        to_port: 443
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ocp_infra_sg
+
+- name: Create an Internet Gateway for the new VPC
+  ec2_vpc_igw:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: present
+    tags:
+      env_id: "{{ env_id }}"
+  when: aws_create_vpc
+  register: new_gw
+
+- name: Create Gateway subnet route table for the new VPC
+  ec2_vpc_route_table:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    tags:
+      Name: "{{ env_id }}-route-table-gw"
+      env_id: "{{ env_id }}"
+    subnets:
+      - "{{ new_subnets.subnet.id }}"
+    routes:
+      - dest: "0.0.0.0/0"
+        gateway_id: "{{ new_gw.gateway_id }}"
+
+# Create local facts with new subnet id
+- name: Register new created objects as facts
+  set_fact:
+    aws_subnet: "{{ new_subnets.subnet.id }}"
+
+- name: Create Master and etcd instances
+  ec2:
+    image: "{{ aws_image_name }}"
+    instance_type: m4.xlarge
```

Comment:
Let's use a variable for the flavor.

Reply:
Ok, but we should use more than one. From the architecture we designed for Labs there were 2 different flavors; one for Masters and Compute Nodes, and another one for Infra Nodes. Should we use 3 so the user could choose a different flavor for Masters, Compute Nodes and Infra Nodes?

Reply:
yes - if you look at the OpenStack implmentation, it supports the following flavors:
`lb_flavor`, `etcd_flavor`, `master_flavor`, `node_flavor`, `infra_flavor`, `dns_flavor`
I would suggest aligning with the same, but fall back on a "default" flavor if not explicitly specified.

Reply:
Sure